### PR TITLE
Huge improvement in detailed statistics

### LINKED ISSUE #56: Efficient Available Dates View
## Description
There currently exists a fairly efficient (not) algorithm for finding the available dates of a certain sensor (as the archive doesn't always take full sensor readings). Unfortunately, the calendar JS we're using only has an event that gets called on every action, which means it's really inefficient to parse a large number of dates at a single time). This has to be improved

## Steps to reproduce
Pick a sensor, go to detailed view
Choose a large number of dates (>100) and press show available dates. Click on the dates and you should see blacked out valued for the unavailable dates. If you try to go back a month, you'll notice that the calendar unfortunately decides to repeat this parsing afterwards, and it'll take a long time to go back a couple of months.

## Additional Information
The design of this is additionally horribly inefficient, and can't just ignore the dates without sensor reading in them (e.g if you wanted to take a large year view of sensor readings)

### LINKED ISSUE #63: Detailed Long Term Pollution View
## Description
Currently, we can view daily dates, however, it'll be quite useful to be able to set up some long term projections and visualizations in the form of detailed views

### PR DESCRIPTION
**Pull request by dambem**
## Relevant issue
#63 
## Description of this PR
Detailed statistics for a while have been super unintuive and only work on single dates. This update improves that so that not only can you choose a multi range of dates easily, you can also see which dates are available (as some dates don't contain information about sensor details). 

Lots of code here

## Checklist
- [x ] Tests all pass

### CODE REVIEW

File: ClimateMonitorV2/views/detailedstats.pug [hunk 1]
```diff
@@ -26,7 +27,12 @@ block content
             div(class="col-md-12")
                 div(id='mapid')
                     div(class="loader")
+        div(class="row")
+            div(class="col-md-12")
+                canvas(id="detailedstats" width="400" height="200")
      script(type='text/javascript' src='/javascripts/config.js')
```
Comment (M-Davies) [inline ClimateMonitorV2/views/detailedstats.pug:33]: You still have references to config.js despite it being deleted

File: ClimateMonitorV2/views/index.pug [hunk 2]
```diff
@@ -100,5 +107,7 @@ block content
      script(type='text/javascript' src='/javascripts/config.js')
```
Comment (M-Davies) [inline ClimateMonitorV2/views/index.pug:123]: And here

**Review by M-Davies — CHANGES_REQUESTED** [review]
A few issues I had when I checked out your branch:
- I get a few errors in the console when running `npm start` (which could be just my environment but it's worth checking).
Screenshot 2020-06-28 at 11 38 37

- Lines [202](https://github.com/dambem/ClimateMonitorV2/blob/master/ClimateMonitorV2/public/javascripts/new_mapping.js#L202), [371-374](https://github.com/dambem/ClimateMonitorV2/blob/master/ClimateMonitorV2/public/javascripts/new_mapping.js#L371-L374) and [382](https://github.com/dambem/ClimateMonitorV2/blob/master/ClimateMonitorV2/public/javascripts/new_mapping.js#L382) still have references to the deleted config.js file which I suspect is causing the 404 in the image

**Comment by dambem** [comment]
Ahh, that's due to config being accidentally deleted when it shouldn't be. Let me just put that back in

**Comment by dambem** [comment]
Could i get an update on this @M-Davies

**Review by M-Davies — CHANGES_REQUESTED** [review]
A few conflicts to fix then it should be ok with me. Don't forget to add the new `GITHUB_API_TOKEN` into the config.js file

**Review by M-Davies — APPROVED** [review]
Looks good to me

**Pull request merged**